libsql-wal/src/checkpointer.rs

```diff
@@ -0,0 +1,425 @@
+use std::future::Future;
+use std::sync::Arc;
+
+use hashbrown::HashSet;
+use libsql_sys::name::NamespaceName;
+use tokio::sync::mpsc;
+use tokio::task::JoinSet;
+
+use crate::io::Io;
+use crate::registry::WalRegistry;
+
+pub(crate) type NotifyCheckpointer = mpsc::Sender<NamespaceName>;
+
+type LibsqlCheckpointer<IO, S> = Checkpointer<WalRegistry<IO, S>>;
+
+impl<IO, S> LibsqlCheckpointer<IO, S>
```

Comment:
I find this really awkward to read with an impl for a type alias, I would maybe suggest newtyping at least?